Client/WebAuthN/WebAuthnClientData.swift

```diff
@@ -0,0 +1,48 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Shared
+
+enum  WebAuthnClientDataType: String {
+    case create = "webauthn.create"
+    case get = "webauthn.get"
+}
+
+struct WebAuthnClientData {
+    var type: String
+    var challenge: String
+    var origin: String
+    
+    enum CodingKeys: String, CodingKey {
+        case type
+        case challenge
+        case origin
+    }
+    
+    init(type: String, challenge: String, origin: String) {
+        self.type = type
+        self.challenge = challenge
+        self.origin = origin
+    }
+}
+
+extension WebAuthnClientData: Encodable {
+    func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(type, forKey: .type)
+        try container.encode(origin, forKey: .origin)
+        
+        let challengeData = Data(base64Encoded: challenge)
+        let websafeChallenge = challengeData?.websafeBase64String()
+        try container.encode(websafeChallenge, forKey: .challenge)
+    }
+}
+
+func clientDataHash (data: Data) -> Data? {
```

Comment:
I don't see this used anywhere?

Reply:
Used here: https://github.com/brave/brave-ios/pull/1158/files#diff-1faf1a9d0dd722c9a8d957734c58b2e6R187

Reply:
Ah didn't notice GitHub didn't unfold that file. Sorry!